cli/src/lockfiles/python.rs

```diff
@@ -88,6 +90,60 @@ impl Parseable for PipFile {
     }
 }
 
+impl Parseable for Poetry {
+    fn new(filename: &Path) -> Result<Self, io::Error>
+    where
+        Self: Sized,
+    {
+        Ok(Poetry(fs::read_to_string(filename)?))
+    }
+
+    /// Parses `poetry.lock` files into a vec of packages
+    fn parse(&self) -> ParseResult {
+        let mut lock: PoetryLock = toml::from_str(&self.0)?;
+
+        // Warn if the version of this lockfile might not be supported.
+        if !lock.metadata.lock_version.starts_with("1.") {
+            log::warn!(
+                "Expected poetry lockfile version ^1.0.0, found {}. \
+                Attempting to continue, but results might be inaccurate.",
+                lock.metadata.lock_version
+            );
+        }
+
+        Ok(lock.packages.drain(..).map(Package::into).collect())
+    }
+}
+
+#[derive(Deserialize, Debug)]
+struct PoetryLock {
+    #[serde(rename = "package")]
+    packages: Vec<Package>,
+    metadata: PoetryMetadata,
+}
+
+#[derive(Deserialize, Debug)]
+struct Package {
+    name: String,
+    version: String,
+}
+
+impl From<Package> for PackageDescriptor {
+    fn from(package: Package) -> Self {
+        Self {
+            name: package.name,
+            version: package.version,
+            package_type: PackageType::PyPi,
+        }
+    }
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "kebab-case")]
+struct PoetryMetadata {
+    lock_version: String,
+}
```

Comment:
I wasn't *completely* sure where to put this. It's *kinda* like a parser, but the `parsers.rs` module is a bit more hand-rolled than this.

I'd personally make the `parsers.rs` file into a directory, extract all the modules into separate files, then put this in `parsers/poetry.rs`. But maybe there's also some better place for it?

Reply:
It's not much code, so I think it's fine to leave it here.

But if you want to make `parsers.rs` into a directory, I'm okay with that, too.

Reply:
Maybe keep it like this for now and save the complete refactor for a later date.